source/production/F0.Minesweeper.Logic/Minefield.cs

```diff
@@ -1,33 +1,98 @@
+using System;
+using System.Collections.Generic;
 using F0.Minesweeper.Logic.Abstractions;
+using F0.Minesweeper.Logic.Mineplacer;
 
 namespace F0.Minesweeper.Logic
 {
 	internal sealed class Minefield : IMinefield
 	{
 		private readonly uint width, height, mineCount;
+		private IMineplacer mineplacer;
+		private bool isFirstUncover = true;
+		private bool[,] minefield;
+		private string DebugVisualization
+		{
+			get
+			{
+				System.Text.StringBuilder sb = new System.Text.StringBuilder();
+				for (int j = 0; j < height; j++)
+				{
+					for (int i = 0; i < width; i++)
+					{
+						sb.Append(minefield[i, j] ? "☒" : "☐");
+					}
+					sb.AppendLine();
+				}
+				return sb.ToString();
+			}
+		}
 
-		internal Minefield(uint width, uint height, uint mineCount)
+		internal Minefield(uint width, uint height, uint mineCount, MinefieldGenerationOptions generationOptions)
 		{
 			this.width = width;
 			this.height = height;
 			this.mineCount = mineCount;
+			switch (generationOptions)
```

Comment:
could be more concise with a _switch expression_

Reply:
As you said, this was a quick-CTRL-Dot-then-Enter-fix. Just lovely! ❤️